src/jdb_to_nwb/convert_photometry.py

```diff
@@ -1,22 +1,218 @@
+    data_size = data_matrix.size # size of the data matrix
+    ID_matrix = eye(data_size,format='csc') # creates an identity matrix the size of data_size in compressed sparse column (csc) format
+    diff_matrix = ID_matrix[1:]-ID_matrix[:-1] # numpy.diff() does not work with sparse matrix. This is a workaround.
+    diagonal_matrix = diags(binary_mask,0,shape=(data_size,data_size)) # creates a diagonal matrix with the binary mask values on the diagonal
+    A = csc_matrix(diagonal_matrix+(lambda_*diff_matrix.T*diff_matrix)) # represents the combination of the diagonal matrix and the smoothness penalty term Lambda_
+    B = csc_matrix(diagonal_matrix*data_matrix.T) # represents the weighted data 
+    smoothed_baseline=spsolve(A,B) # solves the linear system of equations to find the smoothed baseline
+
+    return np.array(smoothed_baseline)
+
+def airPLS(data, lambda_=100, penalty_order=1, max_iterations=15):
+    '''
+    Adaptive iteratively reweighted penalized least squares for baseline fitting
+    DOI : 10.1039/b922045c
+
+    This function is used to fit a baseline to the input data x using adaptive iteratively reweighted penalized least squares.
+    The baseline is fitted by minimizing the weighted sum of the squared differences between the input data and the baseline.
+    The function uses a penalized least squares approach to fit the baseline, with a penalty term that encourages smoothness.
+    The function iteratively adjusts the weights and the baseline until convergence is achieved.
+
+    input
+        data: input data (i.e. chromatogram of spectrum)
+        lambda_: Parameter that can be adjusted by user. The larger lambda is, the smoother the resulting baseline.
+        penalty_order: The order of the penalty for the differences in the baseline. Controls how strongly the smoothness constraint is enforced.
+        max_iterations: Maximum number of iterations for the algorithm to adjust the weights and fit the baseline.
+    
+    output
+        the fitted baseline vector
+    '''
+
+    data_points = data.shape[0] # number of data points
+    weights  = np.ones(data_points) # initial weights set to 1. All points are initially treated equally.
+
+    for i in range(1,max_iterations+1):
+        baseline = WhittakerSmooth(data,weights,lambda_, penalty_order) # loop runs 'max_iterations' times to adjust the weights and fit the baseline
+        delta = data - baseline # difference between data and baseline to calculate residuals, how much each data point deviates from the baseline. delta > 0 == peak. delta < 0 == background.
+        sum_of_neg_deltas = np.abs(delta[delta<0].sum()) # how much data is below the baseline. 
+
+        if (sum_of_neg_deltas<0.001*(abs(data)).sum() or i==max_iterations): # convergence check: if sum_of_neg_deltas is less than 0.001 of the total data, or if the maximum number of iterations is reached, the loop breaks.
+            if (i==max_iterations): print('WARING max iteration reached!')
```

Comment:
```suggestion
            if (i==max_iterations): warnings.warn('Max iteration reached!')
```
Also add `import warnings` at the top of the script